rebound-android/src/main/java/com/facebook/rebound/HardwareLayerSpringListener.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+package com.facebook.rebound;
+
+import android.view.View;
+import java.util.ArrayList;
+import java.util.WeakHashMap;
+
+/**
+ * Spring Listener with Hardware Layer implementation, it's useful if
+ * you use with complex transition in your views.
+ */
+public class HardwareLayerSpringListener implements SpringListener {
+
+  /**
+   * WeakHashMap can avoid any memory leak, needs to be tested.
+   */
+  private WeakHashMap<Integer, View> weakViews = new WeakHashMap<>();
+
+  public HardwareLayerSpringListener(View... views) {
+    addViews(views);
+  }
+
+  public HardwareLayerSpringListener(WeakHashMap<Integer, View> weakViews) {
```

Comment:
This constructor seems unnecessary. I'd prefer to just have the var args version. Sharing a HashMap between a caller and this class seems fragile.